pyproject.toml

```diff
@@ -201,16 +201,22 @@ dev = [
   "python-multipart",
   "psutil",
   # Linting
-  "pylint==2.15.10",
-  # Code formatting
-  "black[jupyter]==22.6.0",
+  "pylint",
+  "farm-haystack[formatting]",
   # Documentation
   "pydoc-markdown",
   "mkdocs",
   "jupytercontrib",
   "watchdog",
   "requests-cache",
 ]
+
+formatting = [
+  # Version specified following Black stability policy:
+  # https://black.readthedocs.io/en/stable/the_black_code_style/index.html#stability-policy
+  "black[jupyter]~=23.0",
```

Comment:
Just to clarify, are we still including `black` in the github hooks? If so we may want to include it as a dependency to `dev` or at least in `all`. If this purely lives in github actions now then this should be fine.

Reply:
Ah right! I forgot about the hook, I didn't update the version there and I think it must set manually. 🤔

Reply:
It looks like we set it here https://github.com/deepset-ai/haystack/blob/c66f855caf55062fa6fdfea133b21cfb11f6355f/.pre-commit-config.yaml#L19

Reply:
Updated, I used a fixed version as recommended by [`pre-commit` docs](https://pre-commit.com/#using-the-latest-version-for-a-repository).